### ISSUE TITLE
Bring back the SourceText field in WarningText

### ISSUE THREAD

**Issue by sjakobi** [sjakobi/haddock#12]

**Comment by sjakobi** [comment]
Relevant, currently failing testcase: T10313

**Comment by sjakobi** [comment]
Relevant history: https://github.com/ghc/ghc/commit/15dd7007275a5dcdae2c9f104773eceaa56590dc

What we might end up with:

```
data WarningTxt name = WarningTxt WarningSort SourceText (WithSourceText (HsDoc name))
```

To parameterize over `name`, we should fix #11 first.

**Comment by sjakobi** [comment]
Fixed in https://github.com/sjakobi/ghc/commit/6a4a818bff28922f80fb2b89dfe8d41fd6803cf1 and https://github.com/sjakobi/ghc/commit/8835bff0bd.
